File: test_feature/companies/consulting/src/test/java/com/github/klee0kai/stone/test_feature/consulting/dependencies/ProvideDependenciesTests.java
package com.github.klee0kai.stone.test_feature.consulting.dependencies;

import com.github.klee0kai.stone.Stone;
import com.github.klee0kai.stone.test_feature.consulting.di.ConsultingComponent;
import com.github.klee0kai.stone.test_feature.hr.di.HrComponent;
import com.github.klee0kai.stone.test_feature.hr.di.HrDependencies;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertNotNull;

public class ProvideDependenciesTests {

    @Test
    void hrNoDepsTest() {
        //Given
        ConsultingComponent appDI = Stone.createComponent(ConsultingComponent.class);
        HrComponent featureDi = Stone.createComponent(HrComponent.class);

        //When
        featureDi.initDependencies(appDI);

        //Then
        HrDependencies hrDeps = featureDi.hrDependencies();
        assertNotNull(hrDeps);
    }

}

File: test_feature/companies/consulting/src/test/java/com/github/klee0kai/stone/test_feature/consulting/dependencies/ProvideDependenciesTests.kt
package com.github.klee0kai.stone.test_feature.consulting.dependencies

import com.github.klee0kai.stone.test_feature.consulting.di.ConsultingComponent
import com.github.klee0kai.stone.test_feature.consulting.di.ConsultingComponentStoneComponent
import com.github.klee0kai.stone.test_feature.finance.di.AccountingComponent
import com.github.klee0kai.stone.test_feature.finance.di.AccountingComponentStoneComponent
import com.github.klee0kai.stone.test_feature.hr.di.HrComponent
import com.github.klee0kai.stone.test_feature.hr.di.HrComponentStoneComponent
import com.github.klee0kai.stone.test_feature.planning.project.LogisticProject
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertNotNull


class ProvideDependenciesTests {

    @Test
    fun hrNoDepsTest() {
        //Given
        val appDI: ConsultingComponent = ConsultingComponentStoneComponent()
        val featureDi: HrComponent = HrComponentStoneComponent()

        //When
        featureDi.initDeps(appDI)

        //Then
        val hrDeps = featureDi.hrDependencies()
        assertEquals(appDI, hrDeps)
    }

    @Test
    fun provideLogisticProjectDepsTest() {
        //Given
        val appDI: ConsultingComponent? = ConsultingComponentStoneComponent()
        val featureDi: AccountingComponent = AccountingComponentStoneComponent()

        //When
        featureDi.initDeps(appDI)

        //Then
        val logisticProject: LogisticProject? = featureDi.dependencies()?.logisticProject()
        assertNotNull(logisticProject)
    }

}
